src/apps/loan-list/list/loan-list.tsx

```diff
@@ -70,67 +108,134 @@ const LoanList: FC<LoanListProps> = ({ pageSize }) => {
     }
   }, [publizonData]);
 
+  const openLoanDetailsModal = useCallback(
+    (modalId: string) => {
+      setModalDetailsId(modalId);
+      open(`${loanDetails}${modalId}`);
+    },
+    [loanDetails, open]
+  );
+
+  const openDueDateModal = useCallback(
+    (dueDateInput: string) => {
+      setDueDate(dueDateInput);
+      open(`${dueDateModal}${dueDateInput}`);
+    },
+    [dueDateModal, open]
+  );
+
   useEffect(() => {
     const modalString = getUrlQueryParam("modal");
+    // if there is a loan details query param, loan details modal should be opened
+    if (modalString && modalString.includes(loanDetails as string)) {
```

Comment:
I think `modalUrlParam`would be a more descriptive than `modalString` that just states that it is a string concerning a modal.

Reply:
done